prboom2/src/MUSIC/portmidiplayer.c

```diff
@@ -185,7 +248,7 @@ static void pm_shutdown (void)
   if (pm_stream)
   {
     // stop all sound, in case of hanging notes
-    pm_stop();
+    // pm_stop() is called on shutdown
```

Comment:
It feels correct to me to call `pm_stop` inside of the shutdown method, even if in practice it can also happen somewhere else. I mean, nothing is stopping us from calling `pm_shutdown` but not shutting down the port, right? Do we need a guard clause in `pm_stop` like `return if not started`? Is it a matter of cleanliness that you are removing the duplication or does it represent a bug?

Reply:
You're right, there's a risk with someone calling `pm_shutdown` directly and causing hanging notes. Calling `pm_stop` twice means sending another couple hundred control changes plus an extra sysex reset. I don't think this matters even with a hardware midi device, but it's not clean like you've said. How about checking `pm_playing` in `pm_shutdown` ? If it's still `1`, then run `pm_stop` first.

Reply:
Thanks 👍